server/api/product.js

```diff
@@ -0,0 +1,63 @@
+const router = require('express').Router()
+const { Product, Review, Category} = require('../db/models')
+
+router.param('productId', function (req, res, next, productId) {
+  Product.findById(productId)
+    .then(product => {
+      req.product = product;
+      next()
+    })
+})
+
+router.get('/', function (req, res, next) {
+  Product.findAll({
+    include: [{
+      model: Category
+    }]
+  })
+    .then(products => res.send(products))
+    .catch(next)
+})
+
+
+router.get('/:productId', function (req, res, next) {
+  Product.findAll({
+    where: { productid: req.params.productId },
+    include: [{ model: Review }]
+  })
+    .then(products => res.send(products))
+    .catch(next)
+})
+
+
+router.delete('/:productId', function (req, res, next) {
+  Product.destroy({ where: { id: req.params.productId } })
+    .then(() => res.sendStatus(204))
+    .catch(next)
+})
+
+router.put('/:productId', function (req, res, next) {
+  Product.findById(req.params.id)
+    .then(function (product) {
+      return product.update(req.body)
```

Comment:
I believe we need a returning: true here because the update function does not automatically return the row it updated